src/common/color/convert-color.ts

```diff
@@ -1,162 +1,163 @@
+ * @returns The RGB array.
+ * @throws If the hex color is invalid.
+ */
 export const hex2rgb = (hex: string): [number, number, number] => {
-  hex = expandHex(hex);
-
+  const color = parse(hex);
+  if (!color || color.mode !== "rgb") {
+    throw new Error(`Invalid hex color: ${hex}`);
+  }
   return [
-    parseInt(hex.substring(0, 2), 16),
-    parseInt(hex.substring(2, 4), 16),
-    parseInt(hex.substring(4, 6), 16),
+    Math.round(color.r * 255),
+    Math.round(color.g * 255),
+    Math.round(color.b * 255),
   ];
 };
 
-export const rgb2hex = (rgb: [number, number, number]): string =>
-  `#${rgb_hex(rgb[0])}${rgb_hex(rgb[1])}${rgb_hex(rgb[2])}`;
-
-// Conversion between LAB, XYZ and RGB from https://github.com/gka/chroma.js
-// Copyright (c) 2011-2019, Gregor Aisch
-
-// Constants for XYZ and LAB conversion
-/* eslint-disable @typescript-eslint/naming-convention */
-const Xn = 0.95047;
-const Yn = 1;
-const Zn = 1.08883;
-/* eslint-enable @typescript-eslint/naming-convention */
-
-const t0 = 0.137931034; // 4 / 29
-const t1 = 0.206896552; // 6 / 29
-const t2 = 0.12841855; // 3 * t1 * t1
-const t3 = 0.008856452; // t1 * t1 * t1
-
-const rgb_xyz = (r: number) => {
-  r /= 255;
-  if (r <= 0.04045) {
-    return r / 12.92;
-  }
-  return ((r + 0.055) / 1.055) ** 2.4;
-};
-
-const xyz_lab = (t: number) => {
-  if (t > t3) {
-    return t ** (1 / 3);
-  }
-  return t / t2 + t0;
-};
-
-const xyz_rgb = (r: number) =>
-  255 * (r <= 0.00304 ? 12.92 * r : 1.055 * r ** (1 / 2.4) - 0.055);
-
-const lab_xyz = (t: number) => (t > t1 ? t * t * t : t2 * (t - t0));
-
-// Conversions between RGB and LAB
-
-const rgb2xyz = (rgb: [number, number, number]): [number, number, number] => {
-  let [r, g, b] = rgb;
-  r = rgb_xyz(r);
-  g = rgb_xyz(g);
-  b = rgb_xyz(b);
-  const x = xyz_lab((0.4124564 * r + 0.3575761 * g + 0.1804375 * b) / Xn);
-  const y = xyz_lab((0.2126729 * r + 0.7151522 * g + 0.072175 * b) / Yn);
-  const z = xyz_lab((0.0193339 * r + 0.119192 * g + 0.9503041 * b) / Zn);
-  return [x, y, z];
+/**
+ * Converts an RGB array to a hex color string.
+ * @param rgb - The RGB array to convert.
+ * @returns The hex color string.
+ * @throws If the RGB array is invalid.
+ */
+export const rgb2hex = (rgb: [number, number, number]): string => {
+  const hex = formatHex({
+    mode: "rgb",
+    r: rgb[0] / 255,
+    g: rgb[1] / 255,
+    b: rgb[2] / 255,
+  });
+  return hex || "#000000";
 };
 
+/**
+ * Converts an RGB array to a LAB array.
+ * @param rgb - The RGB array to convert.
+ * @returns The LAB array.
+ * @throws If the RGB array is invalid.
+ */
 export const rgb2lab = (
   rgb: [number, number, number]
 ): [number, number, number] => {
-  const [x, y, z] = rgb2xyz(rgb);
-  const l = 116 * y - 16;
-  return [l < 0 ? 0 : l, 500 * (x - y), 200 * (y - z)];
+  const labColor = convertRgbToLab({
+    r: rgb[0] / 255,
+    g: rgb[1] / 255,
+    b: rgb[2] / 255,
+  });
+  return [labColor.l, labColor.a, labColor.b];
```

Comment:
Shouldn't we get rid of these legacy types and just start using culori's color objects everywhere? They do include their color space and `[number, number, number]` is very ambiguous